src/DoubleLinkedList.sol

```diff
@@ -100,46 +97,19 @@ library DoubleLinkedList {
         if (list.accounts[id].value != 0) revert AccountAlreadyInserted();
 
         uint256 numberOfIterations;
-        address next = list.head; // If not added at the end of the list `id` will be inserted before `next`.
+        address next = list.accounts[address(0)].next; // If not added at the end of the list `id` will be inserted before `next`.
 
         while (numberOfIterations < maxIterations && next != address(0) && list.accounts[next].value >= value) {
             next = list.accounts[next].next;
             unchecked {
                 ++numberOfIterations;
             }
         }
```

Comment:
this should work too, idk what's best 
```suggestion
        unchecked {
            while (numberOfIterations++ < maxIterations && next != address(0) && list.accounts[next].value >= value) {
                next = list.accounts[next].next;
            }
        }
```

Reply:
To minimize storage reads (most heavy operation), I would have written it like that:

```solidity
uint256 numberOfIterations;
address prevId = address(0);
address nextId = list.accounts[address(0)].next; // If not added at the end of the list `id` will be inserted before `next`.
Account storage nextAccount = list.accounts[nextId];

unchecked {
    while (true) {
        if (++numberOfIterations == maxIterations) {
            nextAccount = list.accounts[address(0)];
            prevId = nextAccount.prev;
            break;
        } else if (nextId == address(0)) {
            nextAccount = list.accounts[address(0)];
            break;
        } else if (nextAccount.value < value) {
            break;
        } else {
            prevId = nextId;
            nextId = nextAccount.next;
            nextAccount = list.accounts[nextId];
        }
    }
}

list.accounts[id] = Account(prevId, nextId, value);
list.accounts[prevId].next = id;
nextAccount.prev = id;
```
<details>
<summary>gas diff</summary>
<img src="https://user-images.githubusercontent.com/108467407/237041456-7b7982d7-4586-4412-ac0f-06a6a9399db4.png">
</details>